src/cliffordsynthesis/encoding/GateEncoder.cpp

```diff
@@ -6,16 +6,17 @@
 #include "cliffordsynthesis/encoding/GateEncoder.hpp"
 
 #include "Encodings/Encodings.hpp"
+#include "operations/OpType.hpp"
 #include "utils/logging.hpp"
 
 namespace cs::encoding {
 
 using namespace logicbase;
 
-void GateEncoder::createSingleQubitGateVariables() {
+void GateEncoder::createSingleQubitGateVariables(const std::size_t tqEncoding) {
```

Comment:
The GateEncoder class should not have to care whether a two-qubit encoding is used. That breaks the separation of concerns that inheritance tries to provide. Overriding the method in the TwoQubitEncoder class is the better solution.
Also, tqEncoding should be a `bool`